main.go

```diff
@@ -232,26 +336,41 @@ func getMilestones(baseURL string, token string, projectID string, state string)
 }
 
 // CreateMilestoneData creates new milestones with title and due date
-func createMilestoneData(advance int, timeInterval string) map[string]milestone {
+func createMilestoneData(advance int, timeInterval string, api string) map[string]milestone {
 	today := time.Now().Local()
 	milestones := map[string]milestone{}
 	switch {
 	case timeInterval == "daily":
 		for i := 0; i < advance; i++ {
 			var m milestone
-			dueDate := today.AddDate(0, 0, i).Format("2006-01-02")
-			title := dueDate
+			var dueDate, title string
+			switch {
+			case api == "gitlab":
+				dueDate = today.AddDate(0, 0, i).Format("2006-01-02")
+				title = dueDate
+			case api == "github":
+				dueDate = today.AddDate(0, 0, i).Format(time.RFC3339)
+				title = today.AddDate(0, 0, i).Format("2006-01-02")
+			}
 			m.Title = title
 			m.DueDate = dueDate
 			milestones[title] = m
 		}
 	case timeInterval == "weekly":
 		for i := 0; i < advance; i++ {
 			var m milestone
+			var dueDate string
 			lastDay := lastDayWeek(today)
 			year, week := lastDay.ISOWeek()
 			title := strconv.Itoa(year) + "-w" + strconv.Itoa(week)
-			dueDate := lastDay.Format("2006-01-02")
+			switch {
+			case api == "gitlab":
+				dueDate = today.AddDate(0, 0, i).Format("2006-01-02")
+				dueDate = lastDay.Format("2006-01-02")
+			case api == "github":
+				dueDate = today.AddDate(0, 0, i).Format(time.RFC3339)
+				dueDate = lastDay.Format(time.RFC3339)
```

Comment:
Looks like there's one of the two lines here that is not used used anymore

Reply:
I think that was a result of having to initialize the dueDate variable outside the switch statement and handling dueDate creation/formatting differently between GitHub and GitLab. Maybe there's a better way to do this though?

Reply:
Well currently dueDate is set to the date of today + `i` days, but then it's set to the last day of the week instead. So the first of the two lines doesn't do anything, it's immediately replaced by the other value.

Reply:
Ok, I made removed the unnecessary lines :)